### PR TITLE
Backup worker

### PR DESCRIPTION
**Pull request by Hirobreak**
- Backup Child Process avoids main thread lag
- Fix next backup timeout if number overflows maximum time

### CODE REVIEW

File: electron_app/src/backup/databaseExport.js [hunk 1]
```diff
@@ -0,0 +1,466 @@
...
+const whereRawEmailQuery = (id, accountId) => `INNER JOIN ${Table.EMAIL_LABEL} 
+    ON ${id} = ${Table.EMAIL_LABEL}.emailId 
+    WHERE ${Table.EMAIL}.accountId=${accountId} 
+    AND ${Table.EMAIL_LABEL}.labelId NOT IN (${excludedLabels.join(',')}) 
+    AND ${Table.EMAIL}.status NOT IN (${excludedEmailStatus.join(',')})
+`;
+
+/* Batches
+----------------------------- */
+const SELECT_ALL_BATCH = 500;
```
Comment (danieltigse) [inline electron_app/src/backup/databaseExport.js:29]: Are you sure to set the same batch for all the tables? The current value for emails is 100.

File: electron_app/src/backup/exporter.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+let password;
+
+const start = async () => {
+  var args = process.argv.slice(2);
+  const dbPath = args[0];
+  const backupPath = args[1];
+  const recipientId = args[2];
+
+  if (!key) {
+    throw new Error(`Database key was never received`);
```
Comment (danieltigse) [inline electron_app/src/backup/exporter.js:32]: are all the `throw error` are catched and printed using the logger?

File: electron_app/src/backup/index.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+const { spawn } = require('child_process');
+const path = require('path');
+
+const exporterPath = path.join(__dirname, 'exporter.js');
+
+const runBackup = (
+  { dbPath, outputPath, key, recipientId, password },
+  progressCallback
+) => {
+  console.log('START BACKUP : ', recipientId);
```
Comment (danieltigse) [inline electron_app/src/backup/index.js:10]: Can we use the logger with all the console.log of this file?

File: email_mailbox/src/components/SettingsAccountBackupWrapper.js [hunk 4]
```diff
@@ -305,7 +305,8 @@ class SettingsAccountBackupWrapper extends Component {
         password,
         notificationParams: export_backup,
         accountObj: {
-          email: myAccount.email
```
Comment (danieltigse) [inline email_mailbox/src/components/SettingsAccountBackupWrapper.js:308]: are you sure about this?

Comment (Hirobreak) [reply email_mailbox/src/components/SettingsAccountBackupWrapper.js:308]: Yes, not needed

File: electron_app/src/backup/databaseExport.js (hunk 1 above)
Comment (Hirobreak) [reply electron_app/src/backup/databaseExport.js:29]: Yes, the fewer queries the better I believe

File: electron_app/src/backup/exporter.js (hunk 2 above)
Comment (Hirobreak) [reply electron_app/src/backup/exporter.js:32]: It is

**Pull request merged**